cmd/zed/zed.d/zed-functions.sh

```diff
@@ -359,6 +363,80 @@ zed_notify_pushbullet()
 }
 
 
+# zed_notify_slack_webhook (subject, pathname)
+#
+# Notification via Slack Webhook <https://api.slack.com/incoming-webhooks>.
+# The Webhook URL (ZED_SLACK_WEBHOOK_URL) identifies this client to the
+# Slack channel. 
+#
+# Requires awk, curl, and sed executables to be installed in the standard PATH.
+#
+# References
+#   https://api.slack.com/incoming-webhooks
+#
+# Arguments
+#   subject: notification subject
+#   pathname: pathname containing the notification message (OPTIONAL)
+#
+# Globals
+#   ZED_SLACK_WEBHOOK_URL
+#
+# Return
+#   0: notification sent
+#   1: notification failed
+#   2: not configured
+#
+zed_notify_slack_webhook()
+{
+    [ -n "${ZED_SLACK_WEBHOOK_URL}" ] || return 2
+
+    local subject="$1"
+    local pathname="${2:-"/dev/null"}"
+    local msg_body
+    local msg_tag
+    local msg_json
+    local msg_out
+    local msg_err
+    local url="${ZED_SLACK_WEBHOOK_URL}"
+
+    [ -n "${subject}" ] || return 1
+    if [ ! -r "${pathname}" ]; then
```

Comment:
prefer "[[" over "[" because it is more robust

Reply:
I'll note that `[[` introduces a bash-ism which this /bin/sh script seems to be otherwise free of. We aren't currently concerned about this script for FreeBSD, but it would be nice to keep the script compatible for distros that don't have bash for /bin/sh, or change the shebang appropriately.

Reply:
We do want to keep this script bash-ism free.

Reply:
[[ is actually ksh-ism, but no worries, as-is will work mostly